### PR TITLE
fix: fix microsoft login issue on iOS after config branch merges

### PR DESCRIPTION
**Pull request by saeedbashir**
### Description

Jira: [LEARNER-10308](https://2u-internal.atlassian.net/browse/LEARNER-10308)

Updated the terminology client ID instead of appID as client ID is also being used by the Microsoft SDK.

### CODE REVIEW

**Comment by rnr** [comment]
@saeedbashir
Maybe the best place to change it is in the configuration repo? Since all teams in the open-source community would have to change their configurations after this key renaming. What do you think?
cc @touchapp @miankhalid

**Comment by saeedbashir** [comment]
> Maybe the best place to change it is in the configuration repo? Since all teams in the open-source community would have to change their configurations after this key renaming. What do you think?

@rnr Android is/was using `CLIENT_KEY` and iOS was using `APP_ID`. It's better to update the config to use the same key for both platforms. And we decided to go with the Android one, as thats the convention the Microsoft SDK itself if using.

**Review by rnr — COMMENTED** [review]
code LGTM

**Comment by rnr** [comment]
I think also needs to change this key in Documentation: https://github.com/edx/edx-mobile-marketplace-ios/blob/2U/develop/Documentation/CONFIGURATION_MANAGEMENT.md

**Comment by saeedbashir** [comment]
> I think also needs to change this key in Documentation: https://github.com/edx/edx-mobile-marketplace-ios/blob/2U/develop/Documentation/CONFIGURATION_MANAGEMENT.md

Done

**Review by rnr — APPROVED** [review]

File: Core/Core/Providers/SocialAuth/MicrosoftAuthProvider.swift [hunk 1]
```diff
@@ -76,10 +76,10 @@ public final class MicrosoftAuthProvider {
     }
 
     private func createClientApplication() throws -> MSALPublicClientApplication {
-        guard let config = Container.shared.resolve(ConfigProtocol.self), let appID =  config.microsoft.appID else {
+        guard let config = Container.shared.resolve(ConfigProtocol.self), let clientID =  config.microsoft.clientID else {
```
Comment (shafqat-muneer) [inline Core/Core/Providers/SocialAuth/MicrosoftAuthProvider.swift]: Line Length Violation: Line should be 120 characters or less; currently it has 122 characters (line_length)

**Review by shafqat-muneer — CHANGES_REQUESTED** [review]

**Review by shafqat-muneer — APPROVED** [review]
LGTM 👍

**Pull request merged**